### PR TITLE
Add user guide for MPI PDQ

### PR DESCRIPTION
**Pull request by bmcd77**
In addition to the overview.md and quickstart.md files 

I've made 2 small changes to the pipeline script export-user-guides.sh.  I hit 2 issues:

- It seemed to just wait first time round (black screen no info). It was wanting the $GITHUB_WORKSPACE env variable to be set. I assumed I only needed to set it if debugging. I've added a message if it is missing or does not exist and  
- It fell over attempting to copy (cp) an empty user-guide directory with a message saying "cannot stat <path>". My 2 files needed to be 1 level lower in the dir structure. Added a simple check for an empty directory so it warns you about an empty directory rather than failing.

Added Geoff & Eugene for interest in technical content writing in case they'd like to comment.

### CODE REVIEW

File: .github/actions/export-user-guides.sh [hunk 1]
```diff
@@ -7,15 +7,26 @@
 
 OUTDIR=user-guides
 
-cd $GITHUB_WORKSPACE
+if [[ -z "$GITHUB_WORKSPACE" ]]
+then
+  echo "env var GITHUB_WORKSPACE is unset or does not exist"
```
Comment (velsietis) [inline .github/actions/export-user-guides.sh:12]:
```suggestion
  echo "env var GITHUB_WORKSPACE is unset or does not exist"
  exit 1
```
the `|| exit` below will exit if the folder doesn't exist, but it the script will still proceed if `GITHUB_WORKSPACE` is unset

File: catalogue/mpi/pdq/user-guide/overview.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+The MPI is by design an [enterprise master patient index](https://en.wikipedia.org/wiki/Enterprise_master_patient_index "Enterprise master patient index") whose aim is to correctly identify patients whenever or wherever they receive treatment. The result of linking patient identity records across a range of information systems is a single ‘gold standard’ identity record for use by new national systems. This helps minimise the number of duplicate records and supports health board system mergers.
+
+### Message Format 
+This Service exposes an HTTP SOAP web service endpoint.
+
+The NHS Wales PDQ (Query Parameter Definition) messages conform to the Integrating the Healthcare Enterprise (IHE) Patient Demographics Query (PDQ) profile [HL7v2.5](https://hl7-definition.caristix.com/v2/HL7v2.5/Segments). 
+
+These consist of a QBP^Q22^QBP^Q21 request and corresponding synchronous RSP^K21 responses as per standard HL7.
```
Comment (velsietis) [inline catalogue/mpi/pdq/user-guide/overview.md:14]: It's technically a `QPB_Q21` Message Structure triggered by a `Q22` event (of type `QBP`) but as discussed, maybe not something for the overview and better placed in a "concepts" section.

File: catalogue/mpi/pdq/user-guide/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+There are a number of different combinations of fields that can be combined to search the MPI. A sample of these will be detailed with examples below. 
+
+## HTTP Header values
+
+To POST any search requests successfully you will need to set the following API Header values. There would be entered under the [Headers tab](https://learning.postman.com/docs/sending-requests/requests/#configuring-request-headers) in Postman or as a [custom HTTP header](https://www.soapui.org/docs/functional-testing/teststep-reference/http-request/headers/) in SOAPUI.
+
+| Field                     |  Value
+|---------------------------|:-------------------------------------------------------------|
+| Ocp-Apim-Subscription-Key |  <your subscription key>*                                    |
```
Comment (velsietis) [inline catalogue/mpi/pdq/user-guide/quickstart.md]:
```suggestion
| Ocp-Apim-Subscription-Key |  \<your subscription key\>*                                    |
```
Need to escape things that look like html in your markdown

File: catalogue/mpi/pdq/user-guide/quickstart.md [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+| SOAPAction                |  http://apps.wales.nhs.uk/mpi/InvokePatientDemographicsQuery |
+| Content-Type              |  text/xml                                                    |
+
+*<your subscription key>* is unique to you and is allocated when you sign up to the [NHS Wales Developer Portal](https://developer.nhs.wales/) and subscribe to sandbox api's under your profile (visible once you have signed in). If you use one of the tryit links on the portal for a sandbox api you will see that it is automatically included in each api call.
```
Comment (velsietis) [inline catalogue/mpi/pdq/user-guide/quickstart.md]:
```suggestion
*\<your subscription key\>* is unique to you and is allocated when you sign up to the [NHS Wales Developer Portal](https://developer.nhs.wales/) and subscribe to sandbox api's under your profile (visible once you have signed in). If you use one of the tryit links on the portal for a sandbox api you will see that it is automatically included in each api call.
```
as above, escaping angle brackets

File: .github/actions/export-user-guides.sh (hunk 1 above)
Comment (bmcd77) [reply .github/actions/export-user-guides.sh:12]: Just tested it that way yes you're correct. Happy to commit that suggestion.

File: catalogue/mpi/pdq/user-guide/overview.md (hunk 2 above)
Comment (bmcd77) [reply catalogue/mpi/pdq/user-guide/overview.md:14]: I can drop line 14 out for now if that would be better? And we could add a task to include a concepts section at a later date. That bit was from pg7 of the document in sharepoint.  

https://nhswales365.sharepoint.com/:w:/r/sites/DigitalHealthEcosystem/_layouts/15/Doc.aspx?sourcedoc=%7B178FD27B-CB8D-4955-A89B-C93C6E3B0AF2%7D&file=Patient%20Demographics%20Query%20Service%20-%20External%20Interface%20Specification%20v1.0.doc&action=default&mobileredirect=true

File: catalogue/mpi/pdq/user-guide/quickstart.md (hunk 3 above)
Comment (bmcd77) [reply catalogue/mpi/pdq/user-guide/quickstart.md]: thumbs up for that one.

File: catalogue/mpi/pdq/user-guide/quickstart.md (hunk 4 above)
Comment (bmcd77) [reply catalogue/mpi/pdq/user-guide/quickstart.md]: As above thumbs up.

File: catalogue/mpi/pdq/user-guide/overview.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+   -     Without GP - PD1 segment of the message is missing
+   -     Without NHS number - missing NHS segment
+
+* Edge Case 2: Response when searching for HUDSON, Martha | 21/03/1939 | F | CF13 9AD | NHS: 5543223455 | PAS(129): E231796
+   -     Deceased patient - returns a PID.29 (Patient Death Date and Time) against the patient
+   -     Multiple matches - multiple RSP_K21.QUERY_RESPONSE segments
+
+NOTE: The mock implementations used by the developer portal will also *echo back the QPD section (request details) of the original request* and populates MSH/MSH.7/TS.1 with a fresh timestamp for each request.
+
+To get started making requests take a look at our [quickstart](/pdq-user-guide/api-guide/quickstart)
```
Comment (velsietis) [inline catalogue/mpi/pdq/user-guide/overview.md]:
```suggestion
To get started making requests take a look at our [quickstart](/api-guides/mpi-pdq/quickstart)
```

File: catalogue/mpi/pdq/user-guide/quickstart.md [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+
+We have included a number of similar searches in our publicly available Postman project. Please see Notes on using Postman below. In a If you are in a hurry 2 things are important:
+
+- you will need to use Postman desktop and 
+- set your developer portal postman <!--{% raw %}-->{{OcpApimSubscriptionKey}}<!--{% endraw %}--> value before a request will be successfull.
+
+[![MPI PDQ Postman project](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/f54d6238f91e83aa9aaf)
+
+In addition we have included 2 edge cases in the Postman search examples. The details of what cases they cover specifically are 
+noted in the [MPI user guide overview](/pdq-user-guide/api-guide/overview)
```
Comment (velsietis) [inline catalogue/mpi/pdq/user-guide/quickstart.md]:
```suggestion
noted in the [MPI user guide overview](/api-guides/mpi-pdq/overview)
```

File: catalogue/mpi/pdq/user-guide/overview.md (hunk 2 above)
Comment (bmcd77) [reply catalogue/mpi/pdq/user-guide/overview.md:14]: Conclusion I think I'll leave that line in there for now. We can always revisit should we add a "concepts" section at a later point.

**Review by BobbyCGD — APPROVED** [review]

**Review by velsietis — COMMENTED** [review]
Looks good

**Pull request merged**